# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../ipt_sysrq/compile-fix-2.6.21.patch
# Copyright (C) 2007 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./src/ipt_SYSRQ-module.c.orig	2007-07-20 14:38:27.000000000 +0300
+++ ./src/ipt_SYSRQ-module.c	2007-07-20 14:51:09.000000000 +0300
@@ -38,12 +38,7 @@
 	if( (c=rsysrq_extract((struct rsysrq_s*)buf,passwd,now.tv_sec,tolerance))==0 )
 		return(0);
 	
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
-	handle_sysrq(c, NULL, NULL, NULL );
-#else
-	handle_sysrq(c, NULL, NULL );
-#endif
-	
+	handle_sysrq(c, NULL);	
 	return(1);
 }
 
@@ -53,8 +48,7 @@
        unsigned int hooknum,
        const struct net_device *in,
        const struct net_device *out,
-       const void *targinfo,
-       void *userinfo)
+	const void *targinfo)
 #else
 static unsigned int
 target(struct sk_buff **pskb,
@@ -107,8 +101,9 @@
 	return 1;
 }
 
-static struct ipt_target ipt_sysrq_reg = {
+static struct xt_target ipt_sysrq_reg = {
         .name           = "SYSRQ",
+	.family		= AF_INET,
         .target         = target,
         .checkentry     = checkentry,
         .me             = THIS_MODULE,
@@ -120,7 +115,7 @@
 
 static int __init init(void)
 {
-	if (ipt_register_target(&ipt_sysrq_reg))
+	if (xt_register_target(&ipt_sysrq_reg))
 		return -EINVAL;
 
 	return 0;
@@ -128,7 +123,7 @@
 
 static void __exit fini(void)
 {
-	ipt_unregister_target(&ipt_sysrq_reg);
+	xt_unregister_target(&ipt_sysrq_reg);
 }
 
 module_init(init);
